src/selection_classic.c

```diff
@@ -86,16 +78,16 @@ void selectionClassicCreate(void)
 
 void selectionClassicDraw(void)
 {
-    const struct Selection *const sel = *selectionGet();
-    const struct SelectionClassic *const pc = sel->classic;
+    const struct Selection *const sel = &selection;
+    const struct SelectionClassic *const pc = &sel->classic;
     XDrawRectangle(disp, root, pc->gc, sel->rect.x, sel->rect.y, sel->rect.w,
         sel->rect.h);
     XFlush(disp);
 }
 
 void selectionClassicMotionDraw(int x0, int y0, int x1, int y1)
 {
-    const struct Selection *const sel = *selectionGet();
+    const struct Selection *const sel = &selection;
 
     if (sel->rect.w)
```

Comment:
This line controls the clearing of the selection.
In order for it to work correctly, you must set the values of `selection.rect` to zero (this was already done by calloc) because otherwise the following happens: `scrot -shide`
![2023-06-11-040304_222x155_scrot](https://github.com/resurrecting-open-source-projects/scrot/assets/73098402/e95bc8bf-6f12-4b6d-a3bd-d6e7b040fe92)
I would recommend to set it to zero in the function: `scrotSelectionCreate`

Reply:
Thanks, fixed now.